============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/chips/aulas/compiladores/trabalhos/lispy-parser/run/160037522
plugins: json-report-1.2.4, metadata-1.11.0, hypothesis-5.41.4
collected 12 items

test_parser.py ..........FF                                              [100%]

=================================== FAILURES ===================================
__ TestAnalisadorSintatico.test_inclui_comando_begin_em_sequencia_de_comandos __

self = <test_parser.TestAnalisadorSintatico object at 0x7f184b7c8fd0>

    def test_inclui_comando_begin_em_sequencia_de_comandos(self):
>       assert value("(cmd 1)\n(cmd 2)") == [
            symbol("begin"),
            [symbol("cmd"), 1],
            [symbol("cmd"), 2],
        ]
E       assert "No terminal defined for '(' at line 2 col 1\n\n(cmd 2)\n^\n" == [Symbol(value='begin'), [Symbol(value='cmd'), 1], [Symbol(value='cmd'), 2]]
E        +  where "No terminal defined for '(' at line 2 col 1\n\n(cmd 2)\n^\n" = value('(cmd 1)\n(cmd 2)')

test_parser.py:127: AssertionError
_____ TestAnalisadorSintatico.test_aceita_programa_completo_com_comentário _____

self = <test_parser.TestAnalisadorSintatico object at 0x7f184b790b50>

    def test_aceita_programa_completo_com_comentário(self):
        src = """
        ;; Fatorial
        (define fat (lambda (n)
            (if (<= n 1)
                1
                (* n (fat (- n 1))))))
    
        (print (fat 5))
        """
        print("Testando comando")
        print(src)
        s = symbol
>       assert value(src) == [
            s("begin"),
            [
                s("define"),
                s("fat"),
                [
                    s("lambda"),
                    [s("n")],
                    [
                        s("if"),
                        [s("<="), s("n"), 1],
                        1,
                        [s("*"), s("n"), [s("fat"), [s("-"), s("n"), 1]]],
                    ],
                ],
            ],
            [s("print"), [s("fat"), 5]],
        ]
E       assert "No terminal defined for ';' at line 2 col 9\n\n        ;; Fatorial\n        ^\n\nExpecting: {'__ANON_0', '__ANON_1', 'A', 'NUMBER', 'QUOTE', 'SYMBOL', 'OP', 'LPAR', 'STRING'}\n" == [Symbol(value='begin'), [Symbol(value='define'), Symbol(value='fat'), [Symbol(value='lambda'), [Symbol(value='n')], [S...bol(value='*'), Symbol(value='n'), [Symbol(value='fat'), [...]]]]]], [Symbol(value='print'), [Symbol(value='fat'), 5]]]
E        +  where "No terminal defined for ';' at line 2 col 9\n\n        ;; Fatorial\n        ^\n\nExpecting: {'__ANON_0', '__ANON_1', 'A', 'NUMBER', 'QUOTE', 'SYMBOL', 'OP', 'LPAR', 'STRING'}\n" = value('\n        ;; Fatorial\n        (define fat (lambda (n) \n            (if (<= n 1)\n                1\n                (* n (fat (- n 1))))))\n\n        (print (fat 5))\n        ')

test_parser.py:147: AssertionError
----------------------------- Captured stdout call -----------------------------
Testando comando

        ;; Fatorial
        (define fat (lambda (n) 
            (if (<= n 1)
                1
                (* n (fat (- n 1))))))

        (print (fat 5))
        
--------------------------------- JSON report ----------------------------------
JSON report written to: .report.json (11982 bytes)
=========================== short test summary info ============================
FAILED test_parser.py::TestAnalisadorSintatico::test_inclui_comando_begin_em_sequencia_de_comandos
FAILED test_parser.py::TestAnalisadorSintatico::test_aceita_programa_completo_com_comentário
========================= 2 failed, 10 passed in 0.22s =========================
